packages/ts/react-signals/src/Signals.ts

```diff
@@ -0,0 +1,78 @@
+import { Signal } from './core.js';
+import type { SetEvent, StateEvent } from './types';
+
+/**
+ * A signal that holds a value. The underlying
+ * value of this signal is stored and updated as a
+ * shared value on the server.
+ *
+ * @internal
+ */
+export abstract class ValueSignal<T> extends Signal<T> {
+  private readonly publish: (event: StateEvent) => Promise<boolean>;
```

Comment:
We shouldn't rely on TS privacy when we have JS privacy
```suggestion
  readonly #publish: (event: StateEvent) => Promise<boolean>;
```